core/src/main/scala-2.13+/cats/data/NonEmptyLazyList.scala

```diff
@@ -0,0 +1,455 @@
+
+  /**
+    * Appends the given LazyList
+    */
+  final def appendLazyList[AA >: A](nell: LazyList[AA]): NonEmptyLazyList[AA] =
+    if (nell.isEmpty) value
+    else create(toLazyList ++ nell)
+
+  /**
+    * Alias for `appendLazyList`
+    */
+  final def :++[AA >: A](c: LazyList[AA]): NonEmptyLazyList[AA] =
+    appendLazyList(c)
+
+  /**
+    * Prepends the given LazyList
+    */
+  final def prependLazyList[AA >: A](c: LazyList[AA]): NonEmptyLazyList[AA] =
+    if (c.isEmpty) value
+    else create(c ++ toLazyList)
+
+  /**
+    * Prepends the given NonEmptyLazyList
+    */
+  final def prependNell[AA >: A](c: NonEmptyLazyList[AA]): NonEmptyLazyList[AA] =
+    create(c.toLazyList ++ toLazyList)
+
+  /**
+    * Alias for `prependNell`
+    */
+  final def ++:[AA >: A](c: NonEmptyLazyList[AA]): NonEmptyLazyList[AA] =
+    prependNell(c)
+
+  /**
+    * Converts this NonEmptyLazyList to a `NonEmptyList`.
+    */ // TODO also add toNonEmptyLazyList to NonEmptyList?
+  final def toNonEmptyList: NonEmptyList[A] =
+    NonEmptyList.fromListUnsafe(toLazyList.toList)
+
+  /**
+    * Converts this LazyList to a `NonEmptyVector`.
+    */
+  final def toNonEmptyVector: NonEmptyVector[A] =
+    NonEmptyVector.fromVectorUnsafe(toLazyList.toVector)
+
+  /**
+    * Returns the head and tail of this NonEmptyLazyList
+    */
+  final def uncons: (A, LazyList[A]) =
+    (head, tail)
+
+  /**
+    * Returns the first element
+    */
+  final def head: A = uncons._1
+
+  /**
+    * Returns all but the first element
+    */
+  final def tail: LazyList[A] = uncons._2
+
+  /**
+    * Tests if some element is contained in this NonEmptyLazyList
+    */
+  final def contains(a: A)(implicit A: Eq[A]): Boolean =
+    toLazyList.contains(a)
+
+  /**
+    * Tests whether a predicate holds for all elements
+    */
+  final def forall(p: A => Boolean): Boolean =
+    toLazyList.forall(p)
+
+  /**
+    * Tests whether a predicate holds for at least one element of this LazyList
+    */
+  final def exists(f: A => Boolean): Boolean =
+    toLazyList.exists(f)
+
+  /**
+    * Returns the first value that matches the given predicate.
+    */
+  final def find(f: A => Boolean): Option[A] =
+    toLazyList.find(f)
+
+  /**
+    * Returns a new `LazyList` containing all elements where the result of `pf` is final defined.
+    */
+  final def collect[B](pf: PartialFunction[A, B]): LazyList[B] =
+    toLazyList.collect(pf)
+
+  /**
+    * Finds the first element of this `NonEmptyLazyList` for which the given partial
+    * function is defined, and applies the partial function to it.
+    */
+  final def collectLazyList[B](pf: PartialFunction[A, B]): Option[B] = toLazyList.collectFirst(pf)
+
+  /**
+    * Like `collectFirst` from `scala.collection.Traversable` but takes `A => Option[B]`
+    * instead of `PartialFunction`s.
+    */ // TODO manually implemented because LazyList doesn't have collectFirstSome
+  final def collectFirstSome[B](f: A => Option[B]): Option[B] =
```

Comment:
We probably don't need to implement this one. I think the default implementation in type class should be fine.